### DISCUSSION TITLE
React Modals

### DISCUSSION THREAD

**Discussion by catakul** [neuefische/questions#46] [Web] [answered]
Hello,

ich hab ein Problem damit popup Fenster mit React Modals anzeigen zu lassen.
Ich möchte das bei einem Klick auf eine Element Kachel sich ein popup Fenster öffnet, welche wiederum individuell mit Objekten aus meiner JSON Datei befüllt werden soll.

Das Problem: 
Es passiert einfach gar nichts wenn ich auf ein Element klicke und ich kann nicht nachvollziehen wo sich das default pop-up Fenster öffnet bzw. wo das Modal zum Einsatz kommt.

[Siehe User Story 3](https://github.com/users/catakul/projects/1/views/1)

**Code snippet aus einer PeriodicTable.js:**

```
const PeriodicTable = () => {  
  const [infos, setInfos] = useState(null)
 const [isOpen, setIsOpen] = useState(false)
 
    function handleClick(elementNumber){ 
         
const elementToFind = data.elements.find((element) => element.number === elementNumber )
setIsOpen(true); setInfos(elementToFind)
 } 
 console.log(infos) 
 
 
    return ( 
      <>
      <div className="periodic-table">
        {data.elements.map((element, index) => ( 
          <button type="button"
          onClick={() => handleClick(element.number) }
          data-propertiesid={element.number}
            className="element"
            key={index}
            style={{
              gridRow: element.ypos,
              gridColumn: element.xpos,
              backgroundColor: colorMap[element.category],
            }}
          >
            <span className="symbol">{element.symbol}</span>
            <span className="number">{element.number}</span>
            <span className="name">{element.name}</span>
          </button>
         
        ))}
      </div>
      <Modal open={isOpen} onClose={() => setIsOpen(false)}>Test Modal
      {infos}
      {/* content={infos} */}
      </Modal>
      </>
    );
  };
  
  export default PeriodicTable;
```

**Code snippet aus meiner Modal.js:**

```
export default function Modal({open, onClose, content}) {
    // if (!open) return null
    // ({open, onClose, content})
 
  return (
    open && (
    <StyledModal >
        <button onClick={onClose}>X</button>
   {/* {content.number} */}
       </StyledModal>
        )
  )
}
```

**Comment by ThomasF85** [comment]
Hallo,

du hast das Grundproblem schon erkannt (das entnehme ich deinem edit dieser Frage) und alles was gerendert werden soll in dein return statement geschrieben - auch dein Modal. Deine handleClick function ändert jetzt nur noch den State und gibt keine Komponente mehr zurück. Das ist alles richtig so.
Wenn ich deinen Code so übernehme bekomme ich ein Modal mit dem Button angezeigt. Ich weiß nicht was deine StyledModal Komponente ist, aber wenn du diese durch zB eine Section ersetzt wird das Modal angezeigt.

**Comment by catakul** [comment, marked as answer]
Hey,

danke für die Antwort! :)

meine StyledModal component ist für das Styling zuständig, (work in progress..)
```
export const StyledModal = styled.div`
  /* display: flex;
  text-align: center;
  flex-flow: column nowrap;
  justify-content: space-between;
  position: fixed; */
  z-index: 2;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 350px;
  min-width: 350px;
  height: 50px;
  background-color: rgba(24, 24, 24, 0.95);
  color: white;
`;
```

Aber wie meinst du das mit der Section? wie bring ich die am besten zum Einsatz.
Ich versuche jetzt halt die Obejekte aus meiner JSON Datei in das Modal einzubetten. Jedoch weiß ich nicht wie, siehe auskommentierte Zeilen in meinem Code snippet.

**Reply by ThomasF85** [reply]
Die prop **content** muss an das Modal übergeben werden (in deiner PeriodicTable Komponente). Dann kannst du in der Modal Komponente auch auf diese prop zugreifen und z.B. `{content.number}` in deinem JSX verwenden.
